### DISCUSSION TITLE
Locked property value changing on button click.

### DISCUSSION THREAD

**Discussion by amit-dalal** [livewire/livewire#5953] [7. Bugs (archived)]
**Locked Property value changed via calling action on button click**

**HomePage.php**

`<?php

namespace App\Livewire\Components;

use Livewire\Attributes\Title;
use Livewire\Attributes\Locked;
use Livewire\Component;

#[Title('Home Page')] 
class HomePage extends Component
{
    #[Locked] 
    public $heading = 'Heading-1';

    public function changeHeading()
    {
        $this->heading = 'New Heading-2';
    }
}
`

**home-page.blade.php**

`<div class="flex flex-col items-center justify-center flex-grow max-w-full max-h-full min-w-full min-h-full overflow-scroll bg-sky-100">
    <input type="text" wire:model.live="heading">
 
    Heading text is: {{ $heading }}

    <button wire:click='changeHeading' class="px-2 py-1 text-sm text-white bg-sky-800">Change Heading</button>
</div>`

Steps:

1. After loading this full page component, browser will display **_Heading text is: Heading-1_**
2. Instead of doing anything else, directly click on button "Change Heading"
3. Locked property value will change and browser will display _**Heading text is: New Heading-2**_
4. It clearly indicates that value of locked property **heading** changed.

**Comment by joshhanley** [comment]
@amit-dalal thanks for reporting! That behaviour is correct. Locked only stops the property from being able to be changed from the front end, using wire:model or any of the JavaScript api. It doesn’t stop you, the developer, from changing it on the backend through code like you provided. 

Hope this helps!

**Reply by marufsharia** [reply]
@joshhanley Regarding your answer, the lock value applied to the element not only prevents changes but also ensures that it remains unmodifiable through the browser console. This adds an extra layer of security to maintain the integrity of the data. am i right ?

**Reply by amit-dalal** [reply]
@joshhanley
In following scenario, I am not able to change locked property value from backend also using same code:
Step 1: Load full page component and browser will display **_Heading text is: Heading-1_**
Step 2: Type some text in input box and it will display exception **_Cannot update locked property: [heading]_**
Step 3: Close the exception pop-up, click on button "**_Change Heading_**" and it will again display exception **_Cannot update locked property: [heading]_** rather than changing value from backend.
As per your reply, value of public property **heading** should have been changed from backend but it didn't happened, **is this behaviour is also correct**?

**Reply by joshhanley** [reply]
@marufsharia yep that's correct.

@amit-dalal can you provide me exact code you are using? As you should be able to modify a locked property from the backend only.

**Reply by amit-dalal** [reply]
@joshhanley I have created a laravel livewire project and then updated livewire to version 3.
Code of Livewire component "HomePage" is same as mentioned in start of discussion:
1. HomePage.php
2. home-page.blade.php
Do you need complete project?

**Reply by romansh** [reply]
@amit-dalal,  there is not any input box in the first post.  
>Step 2: Type some text in input box and it will display exception Cannot update locked property: [heading]

**Reply by joshhanley** [reply]
@amit-dalal no, just the full class and blade view. If what you have shared is already the complete code, then the reason it's not working, will be because you don't have one root element in your view. Try wrapping everything in a `<div>` and then see if it runs 🙂

But you did mention an input and as @romansh said, there is no input in the code you supplied. So please share your component's full class and view so we can have a look.

**Reply by amit-dalal** [reply]
@joshhanley Sorry for the late response. 
Please find below PHP File Code and Blade File Code:

```<?php

namespace App\Livewire\Components;

use Livewire\Attributes\Title;
use Livewire\Attributes\Locked;
use Livewire\Component;

#[Title('Home Page')] 
class HomePage extends Component
{
    #[Locked] 
    public $heading = 'Heading-1';

    public function setHeading()
    {
        $this->heading = 'Heading-2';
    }
}



<div class="flex flex-col items-center justify-center flex-grow max-w-full max-h-full min-w-full min-h-full overflow-scroll bg-sky-100">
    <input type="text" wire:model.live="heading">
 
    Heading text is: {{ $heading }}

    <button wire:click='setHeading' class="px-2 py-1 text-sm text-white bg-sky-800">Set Heading</button>
</div>

**Comment by calebporzio** [comment]
Yeah, `#[Locked]` only protects mutation from the browser (through something like `wire:model`), not from methods that mutate it.

If this seems confusing, I'd welcome a documentation PR that adds a little tip on the locked attribute page.

Thanks
